lib/src/main/java/com/knuddels/jtokkit/TokenEncoder.java

```diff
@@ -85,57 +89,55 @@ private static int getMinRankIndex(List<Integer> ranks) {
         return minRankIndex;
     }
 
-    private static int getNextIndex(List<Integer> ranks, int nextIndex) {
+    private static int getNextIndex(IntArrayList ranks, int nextIndex) {
         while (nextIndex < ranks.size() && ranks.get(nextIndex) == DUMMY_RANK) {
             nextIndex++;
         }
         return nextIndex;
     }
 
-    private static int getPreviousIndex(List<Integer> ranks, int previousIndex) {
+    private static int getPreviousIndex(IntArrayList ranks, int previousIndex) {
         while (previousIndex >= 0 && ranks.get(previousIndex) == DUMMY_RANK) {
             previousIndex--;
         }
         return previousIndex;
     }
 
-    int addTokensAndGetCount(int maxTokenCount, boolean keepEncodings, byte[] utf8Bytes, List<Integer> out, ArrayList<Integer> ranks) {
-        ByteArrayWrapper match = new ByteArrayWrapper(utf8Bytes);
+    int addTokensAndGetCount(int maxTokenCount, boolean keepEncodings, byte[] byteArray, IntArrayList out, IntArrayList ranks) {
+        ByteArrayWrapper match = new ByteArrayWrapper(byteArray);
         int encoded = encode(match);
```

Comment:
for very large tokens (i.e. when `calculateTokensLarge` is selected) this encoding will already cheaply return `MAX_RANK` without hashing it needlessly